account_invoice_production_lot/models/account_move_line.py

```diff
@@ -19,11 +19,14 @@ class AccountMoveLine(models.Model):
     @api.depends("move_line_ids")
     def _compute_prod_lots(self):
         for line in self:
-            line.prod_lot_ids = line.mapped("move_line_ids.move_line_ids.lot_id")
+            line.prod_lot_ids = line.mapped(
+                "sale_line_ids.move_ids.move_line_ids.lot_id"
+            )
 
     def lots_grouped_by_quantity(self):
         lot_dict = defaultdict(float)
-        for sml in self.mapped("move_line_ids.move_line_ids"):
-            if sml.lot_id:
+        for aml in self:
+            smls = aml.mapped("sale_line_ids.move_ids.move_line_ids")
+            for sml in smls:
                 lot_dict[sml.lot_id.name] += sml.quantity
```

Comment:
```suggestion
        for sml in self.mapped("sale_line_ids.move_ids.move_line_ids"):
            lot_dict[sml.lot_id.name] += sml.quantity
```
to make it even more simple